beacon/goclient/attest.go

```diff
@@ -25,6 +27,8 @@ func (gc *GoClient) AttesterDuties(ctx context.Context, epoch phase0.Epoch, vali
 		return nil, fmt.Errorf("attester duties response is nil")
 	}
 
+	recordRequestDuration(gc.ctx, "AttesterDuties", gc.client.Address(), http.MethodPost, time.Since(start))
```

Comment:
It's a good suggestion, although it cannot be applied everywhere, since in some of the places the method/function has more logic implemented and we do not want to capture these, since the metric suppose to indicate the beacon request durations only, hence recording it at the exit would not work.
It was tricky question (we have discussion in the comments) whether to record metric if there is an error from the lib that wraps beacon calls, since errors aren't always associated with the network calls, so that's another reason `defer`s were not used

Reply:
Yeah, I understand it's probably not gonna work for every case we have, but I think it's a good default-solution to use when it applies.

> It was tricky question (we have discussion in the comments) whether to record metric if there is an error from the lib that wraps beacon calls, since errors aren't always associated with the network calls, so that's another reason defers were not used

we can handle `error` in `defer` too however we want if we define `err error` prior to that `defer` and just check `err != nil` in there - it's a bit "ugly" but still better in most cases IMO because there is that clear separation of business logic and measuring/metrics logic

Reply:
We took a slightly different approach: the function that records the metric is now called immediately after the HTTP call to the beacon. This function now also takes an error as an argument. Its implementation has gained some additional responsibilities, such as checking if the error is an API error and, if so, adding the error status code as a metric attribute.